app/javascript/packages/verify-flow/hooks/use-initial-step-validation.ts

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react';
+import type { Dispatch } from 'react';
+import { FormStep } from '@18f/identity-form-steps';
+import useSessionStorage from './use-session-storage';
+
+/**
+ * Returns the index of the given step name in the form steps order.
+ *
+ * @param stepName Step name.
+ * @param steps Steps order.
+ *
+ * @return Step index.
+ */
+const getStepIndex = (stepName: string, steps: FormStep[]) =>
+  steps.findIndex((step) => step.name === stepName);
+
+/**
+ * React hook which validates the expected initial step to present the user, based on past
+ * completion and presence of a URL path fragment. Behaves similar to a useState hook, where the
+ * return value is a tuple of the validated initial step, and a setter for assigning a completed
+ * step.
+ *
+ * @param basePath Path to which the current step is appended to create the current step URL.
+ * @param steps Steps order.
+ *
+ * @return Tuple of the validated initial step and a setter for assigning a completed step.
+ */
+function useInitialStepValidation(basePath: string, steps: FormStep[]): [string, Dispatch<string>] {
+  const [completedStep, setCompletedStep] = useSessionStorage('completedStep');
+  const initialStep = useMemo(() => {
+    const pathStep = window.location.pathname.split(basePath)[1]?.replace(/\W/g, '');
```

Comment:
As discussed in our team's working session today, we may want to have some flexibility to support subpaths for steps, like the path `/verify/v2/password_confirm/forgot_password` being introduced in #6323.

Consider something like:

```suggestion
    const pathStep = window.location.pathname.split(basePath)[1]?.split('/').filter(Boolean)[0];
```

Reply:
Updated in 12fa56378.